### ISSUE TITLE
Invalid Refresh token with local emulator and Identity Platform

### ISSUE THREAD

**Issue by Stf-F** [firebase/firebase-js-sdk#6125]
**Environment**:

  * Operating System version: Mac OS 12.1, M1
  * Browser version: Edge  100.0.1185.29 
  * Firebase SDK version: 9.6.2
  * Firebase Product: Auth + Identity platform 
  * Node 16.13.1, npm 8.1.2 (via nvm use 16)
  * Vue 3 + Vite

**Problem**:
I currently upgraded a perfectly working Firebase Auth project to Identity Platform to use tenancy.
This, however does not play well _at all_ with the local emulators: 

1.  Registrations against a tenant can no longer be seen in the local emulator
2. It is impossible to use the getIdToken(true) method without crashing the front-end 

This problem only exists in localhost but is kind of blocking as the getIdToken(true) is used in core journeys of the web app. Without it, it becomes impossible to progress through the steps of the onboarding flow and heavily impedes the dev and testing flow.  

#### Steps to reproduce:
- In a Firebase project with firebase Auth working, upgrade to Identity Platform  
- Generate a sign in link on the server side via the admin SDK with the tenant ID ( [as per the official docs](https://cloud.google.com/identity-platform/docs/multi-tenancy-managing-tenants#generating_email_action_links))
- Authenticate with the link
- (Update the claims on the server side)
- Invoke the `getIdToken(true)` to force refresh the token
- Watch the world burn 

![image](https://user-images.githubusercontent.com/52630104/162050548-add3161d-ee12-4afe-8dd2-3323d12efb7b.png)

#### Relevant Code:

```javascript
 const getCurrentUser = async (): Promise<Auth["currentUser"]> => {
    return new Promise((resolve, reject) => {
      const unsubscribe = onAuthStateChanged(
        auth,
        async (user) => {
          if (user) {
            if (document.referrer.includes("stripe") ) {
              try {
                await user?.getIdToken(true); // <------ This crashes the frontend
              } catch (e) {
                throw e;
              }
            }
            state.isAuthenticated.value = true;
            state.user.value = user;
            try {
              const { claims } = await user.getIdTokenResult();
              state.claims.value = claims;
              if (typeof claims.roles === "string") {
                console.log("INSIDE");
                if (claims.active && claims.roles.includes("account_owner")) {
                  state.isActive.value = true;
                }
              }
            } catch (e) {
      
              if (e instanceof Error) {
                throw new Error();
              }
            }
          }
          unsubscribe();
          resolve(user);
        },
        (e) => {
          if (e instanceof Error) {
            state.error.value = e.message;
            logClientError(e as Error);
          }
          reject(e);
        }
      );
    });
  };

```

**Comment by Stf-F** [comment]
In hindsight, I realise that this bug might be better raised in the firebase-tools repo rather than here as there does not seem to be errors on live environments. I will therefore create an issue there. Feel free please to close this issue if irrelevant. Thanks. For anyone interested in this issue, it can be found [here](https://github.com/firebase/firebase-tools/issues/4414).

**Comment by did** [comment]
hey @Stf-F, I might have a similar issue in upgrading my working Vue3 + Firebase (last version of the SDK) app to Vite.
My test suite, previously on Jest and now based on Vitest is broken. 
Without going into much detail, each test initializes a new Firebase app connected to the local emulator and delete it when it's done. Btw, I generate a different Firebase projectId for each of them.
The very first test passes but the others fail because the Firebase SDK refuses to create a new Firebase app with the same name and a different config (the projectIds are different.
Digging into the Firebase SDK code and debugging my tests, it turns out that the internal and global Firebase _apps holding all the references to the apps created by the SDK is not consistent during a single test. 
In other words, calling the Firebase `getApps` function from different files in my test doesn't return the same object.
Obviously, this behavior doesn't happen with Webpack + Jest.
So, there is definitively inconsistency with Firebase and Vite which could explain your issue too.
